phoenixRest/views/agenda/__init__.py

```diff
@@ -52,13 +52,48 @@ def create_agenda_entry(context, request):
         return {
             "error": "Event not found"
         }
+
+
     parsed_time = datetime.fromtimestamp(int(request.json_body['time']))
+    parsed_newTime = datetime.fromtimestamp(int(request.json_body['newTime']))
 
     entry = AgendaEntry(title=request.json_body['title'],
                         description=request.json_body['description'],
                         time=parsed_time, 
-                        event=event)
+                        event=event,
+                        sticky=sticky,
+                        location=location,
+                        newTime=parsed_newTime,
+                        newLocation=newLocation,
+                        modifyReason=modifyReason
+                        )
     request.db.add(entry)
     request.db.flush()
     return entry
 
+@view_config(context=AgendaViews, request_method='PATCH', renderer='json', permission='create')
+@validate(json_body={'uuid': str, 'title': str, 'description': str, 'event_uuid': str, 'newLocation': str, 'newTime': int, 'modifyReason': str, 'sticky': bool})
+def modify_agenda_entry(context, request):
+    event = request.db.query(Event).filter(Event.uuid == request.json_body['event_uuid']).first()
+    entry = request.db.query(AgendaEntry).filter(AgendaEntry.uuid == request.json_body['uuid']).first()
+
+    if not event:
+        request.response.status = 404
+        return {
+            "error": "Event not found"
+        }
+
+    parsed_new_time = datetime.fromtimestamp(int(request.json_body['newTime']))
+
+    if entry:
```

Comment:
Here you are doing the opposite of the rest of the codebase I suggest you study how other view handlers are doing it.

Also see this video: https://www.youtube.com/watch?v=CFRhGnuXG-4

Reply:
Fixed.

[Check this commit for the updated version!](https://github.com/phoenixlan/phoenixapi-v1/commit/78e25f671d1f2900f4be89c0e8c76a7f5a3b7cb1?diff=split)